x/dex/keeper/core.go

```diff
@@ -118,6 +109,8 @@ func (k Keeper) DepositCore(
 		amounts1Deposited[i] = inAmount1
 		totalAmountReserve0 = totalAmountReserve0.Add(inAmount0)
 		totalAmountReserve1 = totalAmountReserve1.Add(inAmount1)
+		balance0 = balance0.Sub(inAmount0)
+		balance1 = balance1.Sub(inAmount1)
```

Comment:
what's going on here?

Reply:
This is for checking the remaining balance after a deposit (midway through the batch).

i.e. balance0 after 1st deposit = balance0 - first inAmount0
balance0 after 2nd deposit = balance0 - (first inAmount0) - (second inAmount0)
etc.

Reply:
Doesn't the bank account for whether a user has enough tokens for the transaction?

Reply:
the transfer is only made at the end of the transaction so bank balances arent updated until then.

Reply:
Yes, my point: do we need validation here?